Ginger/GingerCoreNET/DiameterLib/DiameterUtils.cs

```diff
@@ -0,0 +1,2302 @@
+        {
+            mAct.Error += $"{error}{Environment.NewLine}";
+        }
+        public void SaveRequestToFile(bool isSaveRequest, string saveDirectory)
+        {
+            if (isSaveRequest)
+            {
+                RequestFileContent = CreateMessageRawRequestResponse(mAct, Message);
+                string fullFilePath = SaveToFile("Request", RequestFileContent, saveDirectory, mAct);
+                mAct.AddOrUpdateReturnParamActual("Saved Request Filename", Path.GetFileName(fullFilePath));
+            }
+        }
+        public void SaveResponseToFile(bool isSaveResponse, string saveDirectory, DiameterMessage response)
+        {
+            if (isSaveResponse)
+            {
+                ResponseFileContent = CreateMessageRawRequestResponse(mAct, response);
+                string fullFilePath = SaveToFile("Response", ResponseFileContent, saveDirectory, mAct);
+                mAct.AddOrUpdateReturnParamActual("Saved Response Filename", Path.GetFileName(fullFilePath));
+            }
+        }
+        public void ParseResponseToOutputParams(DiameterMessage response)
+        {
+            mAct.AddOrUpdateReturnParamActual($"Command Code: ", response.CommandCode.ToString());
+            mAct.AddOrUpdateReturnParamActual($"Application Id: ", response.ApplicationId.ToString());
+            mAct.AddOrUpdateReturnParamActual($"Hob-By-Hop: ", response.HopByHopIdentifier.ToString());
+            mAct.AddOrUpdateReturnParamActual($"End-To-End: ", response.EndToEndIdentifier.ToString());
+
+            mAct.AddOrUpdateReturnParamActual($"Request: ", response.IsRequestBitSet.ToString());
+            mAct.AddOrUpdateReturnParamActual($"Proxiable: ", response.IsProxiableBitSet.ToString());
+            mAct.AddOrUpdateReturnParamActual($"Error: ", response.IsErrorBitSet.ToString());
+            mAct.AddOrUpdateReturnParamActual($"Retransmit: ", response.IsRetransmittedBitSet.ToString());
+
+            //AVPs
+            foreach (var avp in response.AvpList)
+            {
+                ParseResponseAvpToOutputParams(avp);
+            }
+
+            AddRawResponseAndRequestToOutputParams();
+        }
+        private void AddRawResponseAndRequestToOutputParams()
+        {
+            mAct.RawResponseValues = ">>>>>>>>>>>>>>>>>>>>>>>>>>> REQUEST:" + Environment.NewLine + Environment.NewLine + RequestFileContent;
+            mAct.RawResponseValues += Environment.NewLine + Environment.NewLine;
+            mAct.RawResponseValues += ">>>>>>>>>>>>>>>>>>>>>>>>>>> RESPONSE:" + Environment.NewLine + Environment.NewLine + ResponseFileContent;
+            mAct.AddOrUpdateReturnParamActual("Raw Request: ", RequestFileContent);
+            mAct.AddOrUpdateReturnParamActual("Raw Response: ", ResponseFileContent);
+        }
+        private void ParseResponseAvpToOutputParams(DiameterAVP avp)
+        {
+            mAct.AddOrUpdateReturnParamActual($"AVP({avp.Name}): ", avp.Value);
```

Comment:
1. show parameter with only AVP name(without word AVP)
2.  in Path show AVPS/Name[num of instances]
3. if child, show in path the parent - AVPS/Parent(s)/..../Name[num of instances]

Reply:
done